delft/sequenceLabelling/models.py

```diff
@@ -345,14 +348,36 @@ def __init__(self, config, ntags=None):
                            name='casing_embedding')(casing_input)
         casing_embedding = Dropout(config.dropout)(casing_embedding)
 
+        feature_count = 1 if not config.features_indices else len(config.features_indices)
+        # print("Feature count: " + str(feature_count))
+
+        # features dimension = number of features x vector size
+        feature_input_dimensions = config.features_vector_size * feature_count
+
+        # layout features input and embeddings
+        features_input = Input(shape=(None, feature_input_dimensions), dtype='float32', name='features_input')
+
+        # feature_output_dimensions = feature_count * config.features_embedding_size
+        # features_embedding = Embedding(input_dim=feature_input_dimensions,
+        #                                output_dim=feature_output_dimensions,
+        #                                mask_zero=True,
+        #                                trainable=True,
+        #                                name='features_embedding')
+        # features_embedding_td = TimeDistributed(features_embedding)(features_input)
+
+        features_embedding_td_dense = TimeDistributed(Dense(config.features_embedding_size, name='feature_embeddings_dense'),
```

Comment:
I implemented slightly differently actually: 
```python 
features_embedding = TimeDistributed(Embedding(input_dim=config.features_vocabulary_size,
                                       output_dim=config.features_embedding_size,
                                       mask_zero=True,
                                       trainable=True,
                                       name='features_embedding'), name="features_embedding_td_1")(features_input)

        features_embedding_out = TimeDistributed(Bidirectional(LSTM(config.features_lstm_units, return_sequences=False)),
                                                 name="features_embedding_td_2")(features_embedding)
```
I used an LSTM and not a Dropout... and the default values are:
   -  DEFAULT_FEATURES_VOCABULARY_SIZE = 12
   - DEFAULT_FEATURES_EMBEDDING_SIZE = 4

@kermitt2 could you maybe check the latest version of the model.py?

Reply:
The question then is what's the motivation of the BidLSTM layer just for the features? It will introduce a lot of new parameters and I doubt it's good to separate the learning that much at channel level (the idea is more to mix the channel inputs and then learn I think) - I would really start simple, first embeddings not trainable, simple Dropout (the purpose of the dropout is to "smooth" the features and avoid having one feature over-used and making the other ones ignored). Then add maybe some complexity and layers progressively and evaluate.

Reply:
You're right. Let's start simple!

Reply:
I resolve this conversation and we can keep commenting on the new version - I still have some questions though